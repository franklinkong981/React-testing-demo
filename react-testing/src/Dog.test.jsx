import React from "react";
import {render} from "@testing-library/react";
import Dog from "./Dog";
import { expect, it } from "vitest";

it('should render', () => {
  render(<Dog name="Rusty" isAdopted={true} />);
});

it('should match snapshot', () => {
  const {asFragment} = render(<Dog name="Rusty" isAdopted={true} />);
  expect(asFragment()).toMatchSnapshot();
});

it('should match snapshot', () => {
  const {asFragment} = render(<Dog name="Rusty" isAdopted={false} />);
  expect(asFragment()).toMatchSnapshot();
});