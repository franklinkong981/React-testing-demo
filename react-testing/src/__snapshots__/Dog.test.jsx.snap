// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="Dog"
  >
    <h1>
      Rusty
    </h1>
    <p>
      Rusty has been adopted!
    </p>
  </div>
</DocumentFragment>
`;

exports[`should match snapshot 2`] = `
<DocumentFragment>
  <div
    class="Dog"
  >
    <h1>
      Rusty
    </h1>
    <button>
      Adopt me!
    </button>
  </div>
</DocumentFragment>
`;
